File: src/portfolio/domain/stock.ts
import Price from "./price";
import moment from "moment";

class Stock {
	name: string;
	prices: Price[];

	constructor(name: string, prices: Price[]) {
		this.name = name;
		this.prices = prices;
	}

	price = (date: Date): number => {
		const found_price = this.prices.find((price) =>
			moment(price.date.value, "DD-MM-YYYY").isSame(moment(date, "DD-MM-YYYY"))
		);
		if (!found_price)
			throw new Error(
				`No price found for ${this.name} in day ${moment(date).format(
					"DD-MM-YYYY"
				)}`
			);

		return found_price.price;
	};
}

export default Stock;

File: models/stock.py
class Stock:
    def __init__(self, name, prices):
        self.name = name
        self.prices = prices

    def price(self, date):
        found_price = [price for price in self.prices if price.date == date]
        if not len(found_price):
            raise Exception("No price found for "+self.name+" in day "+date.strftime("%d/%m/%Y"))
        return found_price[0].price
